dspace-api/src/main/java/org/dspace/discovery/SolrServiceImpl.java

```diff
@@ -1422,13 +1585,150 @@ public int compare(MetadataValue mdv1,MetadataValue mdv2) {
+                // mmm... it could be a template item, skip it
+                return;
+            }
+            // workspaceitem
+            List<String> locations = getCollectionLocations(context, workspaceItem.getCollection());
+            SolrInputDocument doc = new SolrInputDocument();
+
+            doc.addField("lastModified", item.getLastModified());
+            if (workspaceItem.getSubmitter() != null) {
+                doc.addField("read", "ws" + workspaceItem.getSubmitter().getID());
+            }
+
+            List<DiscoveryConfiguration> discoveryConfigurations = SearchUtils
+                    .getAllDiscoveryConfigurations(workspaceItem);
+            addDiscoveryFields(doc, context, item, discoveryConfigurations);
+            addBasicInfoToDocument(doc, Constants.WORKSPACEITEM, workspaceItem.getID(), null, locations);
+
+            String acvalue = DSpaceServicesFactory.getInstance().getConfigurationService()
+                    .getProperty("discovery.facet.namedtype.workspace");
+            if (StringUtils.isBlank(acvalue)) {
+                acvalue = workspaceItem.getTypeText();
+            }
+            String fvalue = acvalue;
+            addNamedResourceTypeIndex(doc, acvalue, fvalue);
+            doc.addField("inprogress.item", item.getUniqueIndexID());
+
+            getSolr().add(doc);
+        } else {
+            // so it is an item in the workflow
+            // get the location string (for searching by collection & community)
+            List<String> locations = getCollectionLocations(context, workflowItem.getCollection());
+            SolrInputDocument doc = new SolrInputDocument();
+
+            doc.addField("inprogress.item", item.getUniqueIndexID());
+            doc.addField("lastModified", item.getLastModified());
+            if (workflowItem.getSubmitter() != null) {
+                addFacetIndex(doc, "submitter", workflowItem.getSubmitter().getID().toString(),
+                        workflowItem.getSubmitter().getFullName());
+            }
+
+            List<DiscoveryConfiguration> discoveryConfigurations = SearchUtils
+                    .getAllDiscoveryConfigurations(workflowItem);
+            addDiscoveryFields(doc, context, item, discoveryConfigurations);
+
+            List<ClaimedTask> claimedTasks = claimedTaskService.find(context, workflowItem);
+            List<PoolTask> pools = poolTaskService.find(context, workflowItem);
+
+            List<SolrInputDocument> docs = new ArrayList<SolrInputDocument>();
+
+            if (claimedTasks != null) {
+                for (ClaimedTask claimedTask : claimedTasks) {
+                    SolrInputDocument claimDoc = doc.deepCopy();
+                    addBasicInfoToDocument(claimDoc, Constants.WORKFLOW_CLAIMED, claimedTask.getID(), null, locations);
+                    claimDoc.addField("workflow.action", claimedTask.getActionID());
+                    claimDoc.addField("workflow.step", claimedTask.getStepID());
```

Comment:
Does this imply, when faceting on the step, the facets will contain the step IDs, and not human-readable step names? Or is a different field used for this purpose?

Reply:
I have reduced the number of fields used as many were redundant. in these field we will have the action and step ID that are "string" so readable but not really ready to be displayed to the user. They are not really needed, nevertheless is quite convenient to have them immediately available in the solr document if you need to perform some analysis / check or manual report